### PR TITLE
save intra-shard scrs as well for dblookup extension

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
save intra-shard miniblocks as well when calling `RecordBlock` - this will also help save the intra shard SCRs that were not available when calling `/transaction/<hash>` endpoint

### CODE REVIEW

File: dblookupext/interface.go [hunk 1]
```diff
@@ -19,6 +20,7 @@ type HistoryRepository interface {
 		blockBody data.BodyHandler,
 		scrResultsFromPool map[string]data.TransactionHandler,
 		receiptsFromPool map[string]data.TransactionHandler,
+		intraShardMiniBlock []*block.MiniBlock,
```
Comment (andreibancioiu) [inline dblookupext/interface.go:23]: Plural, as well.

File: factory/disabled/txCoordinator.go [hunk 2]
```diff
@@ -105,6 +105,11 @@ func (txCoordinator *TxCoordinator) CreateMarshalizedReceipts() ([]byte, error)
 	return nil, nil
 }
 
+// GetCreatedInShardMiniBlocks returns nil as is it disabled
+func (txCoordinator *TxCoordinator) GetCreatedInShardMiniBlocks() []*block.MiniBlock {
```
Comment (andreibancioiu) [inline factory/disabled/txCoordinator.go:109]: `*intrashard*` (naming, optional)

File: node/external/transactionAPI/apiTransactionProcessor.go [hunk 3]
```diff
@@ -210,7 +210,7 @@ func (atp *apiTransactionProcessor) lookupHistoricalTransaction(hash []byte, wit
 	txBytes, txType, found := atp.getTxBytesFromStorageByEpoch(hash, miniblockMetadata.Epoch)
 	if !found {
 		log.Warn("lookupHistoricalTransaction(): unexpected condition, cannot find transaction in storage")
-		return nil, fmt.Errorf("%s: %w", ErrCannotRetrieveTransaction.Error(), err)
+		return nil, ErrCannotRetrieveTransaction
```
Comment (andreibancioiu) [inline node/external/transactionAPI/apiTransactionProcessor.go:213]: :+1:

File: factory/disabled/txCoordinator.go (hunk 2 above)
Comment (bogdan-rosianu) [reply factory/disabled/txCoordinator.go:109]: just kept the naming from interimProcessors

File: dblookupext/historyRepository.go [hunk 4]
```diff
@@ -127,6 +127,7 @@ func (hr *historyRepository) RecordBlock(blockHeaderHash []byte,
 	blockBody data.BodyHandler,
 	scrResultsFromPool map[string]data.TransactionHandler,
 	receiptsFromPool map[string]data.TransactionHandler,
+	intraShardMiniBlock []*block.MiniBlock,
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply dblookupext/historyRepository.go:130]: done

File: dblookupext/interface.go (hunk 1 above)
Comment (bogdan-rosianu) [reply dblookupext/interface.go:23]: done

File: dblookupext/historyRepository.go [hunk 5]
```diff
@@ -127,6 +127,7 @@ func (hr *historyRepository) RecordBlock(blockHeaderHash []byte,
 	blockBody data.BodyHandler,
 	scrResultsFromPool map[string]data.TransactionHandler,
 	receiptsFromPool map[string]data.TransactionHandler,
+	intraShardMiniBlocks []*block.MiniBlock,
```
Comment (AdoAdoAdo) [inline dblookupext/historyRepository.go:130]: can you rename to createdIntraShardMiniBlocks ?

File: dblookupext/interface.go [hunk 6]
```diff
@@ -19,6 +20,7 @@ type HistoryRepository interface {
 		blockBody data.BodyHandler,
 		scrResultsFromPool map[string]data.TransactionHandler,
 		receiptsFromPool map[string]data.TransactionHandler,
+		intraShardMiniBlocks []*block.MiniBlock,
```
Comment (AdoAdoAdo) [inline dblookupext/interface.go:23]: rename here as well?

Comment (bogdan-rosianu) [reply dblookupext/interface.go]: renamed

**Review by andreibancioiu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**